Chronos/Views/AmountsChart.swift

```diff
@@ -11,10 +11,31 @@ import SwiftUI
 import Charts
 import ObservedOptionalObject
 
+public extension Date {
+    /// Returns the start of the interval for the specified calendar component.
+    /// - Parameters:
+    ///   - component: The calendar component (e.g., .day, .weekOfYear, .month, etc.) to truncate to.
+    ///   - calendar: The calendar to use; defaults to Calendar.current.
+    /// - Returns: The start of the interval if available, otherwise the start of the day.
+    func truncated(to component: Calendar.Component, calendar: Calendar = .current) -> Date {
+        guard let interval = calendar.dateInterval(of: component, for: self) else {
+            return calendar.startOfDay(for: self)
+        }
+        return interval.start
+    }
+}
```

Comment:
This should have its own file under `Extenisons`, similar to all the other extensions.